### PR TITLE
ghostie: init at 0.2.1

### PR DESCRIPTION
**Pull request by matthiasbeyer**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/ghostie/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  cargoSha256 = "sha256-YF808suqfeM156KkRGCCtGFsCdgQ4eu6n2P6yAVV7qc=";
+
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  buildInputs = [
+    openssl
+    sqlite
+  ];
```
Comment (figsoda) [inline pkgs/tools/misc/ghostie/default.nix:30]:
```suggestion
  ] ++ lib.optionals stdenv.isDarwin [
    darwin.apple_sdk.frameworks.Cocoa
  ];
```

File: pkgs/tools/misc/ghostie/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  # 4 out of 5 tests are notification tests which do not work in nix builds
+  doCheck = false;
+
```
Comment (figsoda) [inline pkgs/tools/misc/ghostie/default.nix:34]:
```suggestion

  preBuild = lib.optionalString stdenv.isDarwin ''
    export HOME=$(mktemp -d)
  '';

```

File: pkgs/tools/misc/ghostie/default.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  meta = with lib; {
+    description = "Github notifications in your terminal";
+    homepage = "https://github.com/attriaayush/ghostie";
```
Comment (figsoda) [inline pkgs/tools/misc/ghostie/default.nix:37]:
```suggestion
    homepage = "https://github.com/attriaayush/ghostie";
    changelog = "https://github.com/attriaayush/ghostie/releases/tag/v${version}";
```

File: pkgs/tools/misc/ghostie/default.nix [hunk 4]
```diff
@@ -0,0 +1,42 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, stdenv
+, pkg-config
+, openssl
+, sqlite
```
Comment (figsoda) [inline pkgs/tools/misc/ghostie/default.nix:7]:
```suggestion
, pkg-config
, openssl
, sqlite
, stdenv
, darwin
```

**Review by figsoda — CHANGES_REQUESTED** [review]

**Comment by matthiasbeyer** [comment]
@ofborg build ghostie

**Review by figsoda — APPROVED** [review]

**Pull request merged**